### PR TITLE
Build on Appveyor CI

### PR DESCRIPTION
**Pull request by zmughal**
# <strike>Do not merge</strike>
## TODO
- [x] Create a `TMPDIR` like in the [PDL appveyor.yml](https://github.com/PDLPorters/pdl/blob/master/appveyor.yml).

``` yaml
  - mkdir %APPVEYOR_BUILD_FOLDER%\tmp
  - set TMPDIR=%APPVEYOR_BUILD_FOLDER%\tmp
```
- [x] When scanning for the cpanfile prereqs, exclude tools in `dev` and `ci` directories.
- [x] When scanning for the cpanfile prereqs, insert message at top:

``` perl
# Generated by scan-prereqs-cpanfile. DO NOT EDIT!
```
- [x] Try setting `MAKEFLAGS="-j4"` for `Alien::MuPDF` because it uses GNU make.
- [x] <strike>Set `MAXPROCESS` for [dmake-based builds](https://github.com/mohawk2/dmake/blob/master/man/dmake.nc).</strike> Using `-P4` in `MAKEFLAGS` also works.

### CODE REVIEW

**Comment by StayerX** [comment]
Looks good to me.

File: cpanfile [hunk 1]
```diff
@@ -1,6 +1,21 @@
-requires 'Moo';
```
Comment (zmughal) [inline cpanfile:1]: <strike>TODO: need to put a message at top:</strike> DONE

``` perl
# Generated by scan-prereqs-cpanfile. DO NOT EDIT!
```

**Comment by zmughal** [comment]
Ugh, this is really annoying. :-P

File: dist.ini [hunk 2]
```diff
@@ -46,3 +47,6 @@ allow_dirty = dist.ini
 allow_dirty = README
 push_to = origin
 push_to = origin build/master:build/master
+
+[Run::BeforeBuild]
+run = which scan-prereqs-cpanfile && scan-prereqs-cpanfile > cpanfile; true
```
Comment (zmughal) [inline dist.ini]: Exclude tools in dev and ci dirs

**Comment by zmughal** [comment]
@project-renard/dev , this PR is ready for merging.

I have a write-up on the process at https://project-renard.github.io/doc/development/meeting-log/posts/2016/05/03/windows-build-with-msys2/.

**Pull request merged**